File: inst/expt/anqi-bug2.py
#! /usr/bin/env python

from cvxpy import *
import canonInterface
import numpy
import scipy as sp

x = Variable(1)
obj = Maximize(log(x))
prob = Problem(obj)
sol = prob.solve(verbose = True)
data = prob.get_problem_data(ECOS)

d = numpy.matrix([[ 0., -1.],
                  [-1.,  0.],
                  [ 0.,  0.]])

c = numpy.array([ 0., -1.])
G = sp.sparse.csc_matrix(d)
h = numpy.array([-0., -0.,  1.])
dims = {'ep': 1, 'f': 0, 'l': 0, 'e': 1, 'q': [], 's': []}


d1 = numpy.matrix([[ 0., -1.],
                   [ 0.,  0.],
                   [-1.,  0.]])

c1 = numpy.array([ 0., -1.])
G1 = sp.sparse.csc_matrix(d1)
h1 = numpy.array([-0., 1,  0])
dims1 = {'ep': 1, 'f': 0, 'l': 0, 'e': 1, 'q': [], 's': []}

ecos.solve(c1,G1,h1,dims1, verbose = True)

File: inst/expt/anqi-bug2.R
library(cvxr)
x <- Variable(1)
obj <- Maximize(Log(x))
prob <- Problem(obj)
data <- get_problem_data(prob, "ECOS")
base::trace("Solver.solve", tracer = browser, exit = browser, signature = c("ECOS"))
solve(prob, verbose = 1)

cat("c is \n")
print(data$c)
cat("G is \n")
print(data$G)
cat("h is \n")
print(data$h)
cat("dims is\n")
print(data$dims)

data$dims$e <- data$dims$ep
ns <- names(data$dims)
data$dims <- lapply(data$dims, as.integer)
names(data$dims) <- ns

ECOSolveR::ECOS_csolve(data$c, data$G, data$h, data$dims, control = ecos.control(verbose=1L))

## Correct answer, i.e. what cvxpy produces for G, and h. G and h are the only difference!
G1 <- Matrix(c(0., -1.,
               -1.,  0.,
               0.,  0), byrow = TRUE, ncol = 2, sparse = TRUE)

h1 = c(0, 0, 1.0)

ECOSolveR::ECOS_csolve(data$c, G1, h1, data$dims, control = ecos.control(verbose=1L))
